### PR TITLE
Implement colored filenames based on file status

### PR DESCRIPTION
**Pull request by MonliH**
This PR implements coloured filenames based on git and diagnostic status, as requested in #293. This is what it looks like (with a custom colour scheme):
image

I also ran `npx prettier -w  .`, and it formatted a few other files, to my surprise. The filename for `src/git/manager.test.ts` was originally `src/git/manater.test.ts`, which I renamed (typo).

I will document the new highlights added after this gets merged.

### CODE REVIEW

File: src/diagnosticManager.ts [hunk 1]
```diff
@@ -23,6 +24,9 @@ export class DiagnosticManager {
           const errorPathCountNum: Record<string, number> = {};
           const warningPathCountNum: Record<string, number> = {};
 
+          const errorPathsNew: Set<string> = new Set();
+          const warningPathsNew: Set<string> = new Set();
```
Comment (weirongxu) [inline src/diagnosticManager.ts:28]: These variables are never used?

File: src/diagnosticManager.ts [hunk 2]
```diff
@@ -48,6 +54,41 @@ export class DiagnosticManager {
...
+      const parts = relativePath.split(pathLib.sep);
+
+      for (let i = 1; i <= parts.length; i++) {
+        const frontalPath = pathLib.join(
+          root,
+          parts.slice(0, i).join(pathLib.sep),
+        );
+        warningPaths.add(frontalPath);
+      }
+    });
```
Comment (weirongxu) [inline src/diagnosticManager.ts:87]: Why not call `getMixedError` and `getMixedWarning` directly, are they different?

**Review by weirongxu — CHANGES_REQUESTED** [review]

**Comment by weirongxu** [comment]
Thank you very much! :tada:

**Comment by MonliH** [comment]
No problem, I'm looking to contribute again here, `coc-explorer` is an awesome project!

**Pull request merged**